pagemmultiscope_spec.rb

```diff
@@ -0,0 +1,75 @@
+require File.dirname(__FILE__) + '/spec_helper'
+require File.expand_path("../../lib/pagem", __FILE__)
+
+describe PagemMultiscope do
+  let(:scope) { double("Scope") }
+  let(:second_scope) { double("Scope") }
+
+  before(:each) do
+    scope.stub(:size).and_return(101)
+    second_scope.stub(:size).and_return(202)
+    scope.stub(:where => scope, :limit => scope, :offset => scope)
+    second_scope.stub(:where => second_scope, :limit => second_scope, :offset => second_scope)
+  end
+
+  it 'inherits from Pagem' do
+    expect(PagemMultiscope).to be < Pagem
+  end
+
+  it 'gets the sizes of the scopes if they are not passed in' do
+    scope.should_receive(:count).and_return(101)
+    @multipager = PagemMultiscope.new(scope, second_scope, {:page => 1})
+    @multipager.instance_eval { @count}.should == 303
+  end
+
+  it 'assigns of the scopes if they are passed in' do
+    @multipager = PagemMultiscope.new(scope, second_scope, {page: 1}, {
+      count_number: 303,
+      first_scope_count: 101,
+      second_scope_count: 202
+    })
+    @multipager.instance_eval { @count}.should == 303
+  end
+
+  describe "#paged_scope" do
+    before(:each) do
+      scope.should_receive(:count).and_return(101)
+      scope_count = 101
+    end
+
+    context 'current page > 0' do
+
+      context 'end_offset <= first scope count' do
+        it "returns only the first scope" do
+          @multipager = PagemMultiscope.new(scope, second_scope, {page: 1})
+          @start_offset = (@multipager.current_page - 1) * @multipager.items_per_page
```

Comment:
I think this should set <code>multipager.current_page</code> and <code>multiple.items_per_page</code> and let the app code take care of these ivars rather than setting them here